pennylane/pauli/utils.py

```diff
@@ -698,6 +671,26 @@ def is_qwc(pauli_vec_1, pauli_vec_2):
     return True
 
 
+def _are_pauli_words_qwc_pauli_rep(lst_pauli_words):
+    """Given a list of observables assumed to be valid Pauli words, determine if they are pairwise
+    qubit-wise commuting. This private method is used for oeprators that have a valid pauli
+    representation"""
+    basis = {}
+    for op in lst_pauli_words:  # iterate over the list of observables
+        pw = next(iter(op.pauli_rep))
```

Comment:
Is it anywhere ensured that the op is not a PauliSentence with multiple terms? I'd say the following should return False
```
qml.operation.enable_new_opmath()
ls = [
    X(0),
    X(1),
    X(0) + Y(0)
]
>>> qml.pauli.utils._are_pauli_words_qwc_pauli_rep(ls)
True
```

Could that become a problem?

Reply:
do you suspect that users would call `are_pauli_words_qwc` (public function that calls this helper) with things that are not pauli words? My general attitude with arg validation in python is: if code might encourage users to do the wrong thing, provide validation. ofc it's better to not encourage that at all, but sometimes it happens.

Reply:
Yeah I agree that this doesnt need validation for a private function, I was more wondering if this could happen in the overall pipeline of grouping (or otherwhere this function is used on lists of operators) where some operators are sums of paulis whose PauliSentence has multiple terms that would eventually hit this problem

Reply:
This makes me thing, @mudit2812 I think there are no instances of `Sum` operators in the tests, is that missing or because there are steps before hitting group_observables that strips the terms from it?

Reply:
we should probably add validation. it's pretty much called every time we make a measurement with default.qubit

Reply:
This gives me an odd error

```
ops = [
    X(0),
    X(1),
    X(0) + Y(0)
]

qml.pauli.group_observables(ops)
```

```
  File "/Users/korbinian.kottmann/anaconda3/envs/pennylane/lib/python3.11/site-packages/stack_data/core.py", line 660, in executing_piece
    return only(
           ^^^^^
  File "/Users/korbinian.kottmann/anaconda3/envs/pennylane/lib/python3.11/site-packages/executing/executing.py", line 116, in only
    raise NotOneValueFound('Expected one value, found 0')
executing.executing.NotOneValueFound: Expected one value, found 0
```

It says in the docs that `observables` should be a list of pauli word Observables so this is technically fine

Reply:
I ran that on this branch and got the expected error (`TypeError: Expected a Pauli word Observable instance, instead got PauliX(wires=[0]) + PauliY(wires=[0]).`) - that might be related to your environment. Those packages are dependencies of ipykernel, so it might just be an IPython issue with 3.11 or your specific version of it

Reply:
okay great 👍

Reply:
Such validation already exists in `pauli_to_binary`. It has a `check_is_pauli_word` boolean keyword argument which is `True` by default, and if true, then the function validates that the obsrvable is not invalid, hence the error that @timmysilv saw.

Reply:
Thanks for clarifying! If I got this right, and this function assumes valid pauli words (and not sentences) as described in the doc string, then this is actually fine I think 👍 no need to reinvent the wheel here